feat: update readme (#1)
eve-autonomy/go_interface: README.md

> # go_interface

Reviewer: ```suggestion
# Go interface
```

## Overview

> This node integrates user-defined on-demand delivery apps with in-vehicle systems.

Reviewer: ```suggestion
This node integrates user-defined on-demand delivery apps with autonomous driving systems.
```

> The following use cases are realized by using it in combination with FMS (fleet management service) provided by TIER IV, inc.

Reviewer: ```suggestion
Following use-cases are realized by connecting with FMS (fleet management service) providing by TIER IV, inc.
```

> - Switch between on-demand delivery and regular delivery.
> - Call the ego vehicle to any loading / unloading place.
> - Send the ego vehicle to any loading / unloading place.

Reviewer: ```suggestion
- Switch operation mode between on-demand delivery and regular delivery.
- Call an ego vehicle to a designated stop.
- Send an ego vehicle to a designated stop.
```

## Input and Output
- input
  - from [Web.Auto](https://tier4.jp/en/products/#webauto) (DevOps Platform provided by TIER IV)
    - `/webauto/vehicle_info` : Gets unique ID of the ego vehicle managed by Web.Auto.

>   - from [autoware_state_machine](https://github.com/eve-autonomy/autoware_state_machine) 

Reviewer: ```suggestion
  - from [autoware_state_machine](https://github.com/eve-autonomy/autoware_state_machine/) 
```

    - `/req_change_lock_flg` : Receives reservations for on-demand delivery.
  - from [on-demand delivery apps (user-defined)](#required-specifications-for-on-demand-delivery-apps)
    - `GET /api/vehicle_status` (REST API) : Gets the current reservation status for on-demand delivery in the ego vehicle.
- output

>   - to [autoware_state_machine](https://github.com/eve-autonomy/autoware_state_machine)

Reviewer: ```suggestion
  - to [autoware_state_machine](https://github.com/eve-autonomy/autoware_state_machine/)
```
